src/vumi2/transports/smpp/processors.py

```diff
@@ -159,7 +159,10 @@ async def handle_outbound_message(self, message: Message) -> list[SubmitSM]:
         """
         # TODO: support USSD over SMPP
         codec = DataCodingCodecs[self.config.data_coding.name]
-        message_content = (message.content or "").encode(codec.value)
+        try:
+            message_content = (message.content or "").encode(codec.value)
+        except Exception as e:
+            message_content = (message.content or "").encode("utf-8")
```

Comment:
I don't think that this will work. We can only send data encoded with the given configured encoding (defaults to SMSC). If we want to be able to send unicode over smpp, we have to send it as UCS2, that's easily configured in the transport config, in the `data_coding` field.

Keep in mind that UCS2 reduces the amount of characters in an SMS to 70, and so can greatly increase the cost. So what we normally want is fix the content, and replace any non-SMSC characters with something similar that appears in SMSC, and not try to send unicode SMSes.

Previously we've created middleware to do this automatically, to avoid mistakes in the content making their way through to the transport, and then if the desired action is to change the characters to their closest SMSC, then we can just enable that middleware. So it's also a solution to add middleware to vumi2, and then use the unidecode library to create a character replacement middleware, similar to what we have in vumi1: https://github.com/praekeltfoundation/vumi-unidecode-middleware